anchor/anchorparser.go

```diff
@@ -52,6 +54,45 @@ func ParseAnchors(protoAnchors []*yotiprotoattr.Anchor) []*Anchor {
 	return processedAnchors
 }
 
+func parseExtension(ext pkix.Extension) (anchorType Type, val string, err error) {
+	anchorType = AnchorTypeUnknown
+
+	switch {
+	case ext.Id.Equal(sourceOID):
+		anchorType = AnchorTypeSource
+	case ext.Id.Equal(verifierOID):
+		anchorType = AnchorTypeVerifier
+	default:
+		return anchorType, "", errors.New("unknown anchor type")
+	}
+
+	var ae anchorExtension
+	_, err = asn1.Unmarshal(ext.Value, &ae)
+
```

Comment:
It generally make things easier to follow if there's no space between the function call and the error handling.

Reply:
Sure. Is it just your preference on using switch statements here instead of:
```
if err != nil {
    return anchorType, "", fmt.Errorf("unable to unmarshal extension: %v", err)
} else if len(ae.Extension) == 0 {
    return anchorType, "", errors.New("empty extension")
}

val = ae.Extension
```

I've always used the above when we're not switching on a definitive object type

Reply:
They're both pretty much the same (I've not had a dig to see if they are optimised differently). It's a fairly common pattern to use a `switch` for more complex error (or just return values) processing, in my opinion since the logic can be expressed as scenarios (i.e. cases) it's cleaner to write it that way, rather than chaining `if`s. Unless the processing code is hard to follow as written I won't tend to suggest changing between `if`/`switch`.